packages/graphql/src/registry.ts

```diff
@@ -0,0 +1,46 @@
+import { navigateProgram, type Program, type SemanticNodeListener } from "@typespec/compiler";
+import type { GraphQLObjectType } from "graphql";
+
+type Mutable<T> = {
+  -readonly [k in keyof T]: T[k];
+};
+
+// This class contains the registry of all the GraphQL types that are being used in the program
+export class GraphQLTypeRegistry {
+  program: Program;
+  readonly programNavigated: boolean = false;
+
+  constructor(program: Program) {
+    this.program = program;
+    return new Proxy(this, {
+      get(target: GraphQLTypeRegistry, prop: string, receiver) {
+        if (GraphQLTypeRegistry.#publicGetters.includes(prop)) {
+          if (!target.programNavigated) {
```

Comment:
Why would there be a situation where the program has been navigated? `navigateProgram` is deterministic in it's walk, so there is no reason to check for this.

Reply:
See the longer discussion below. Basically, the intent here is that `navigateProgram` is never called explicitly — it's an implementation detail.